lib/engine/ability/purchase_train.rb

```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Ability
+    class PurchaseTrain < Base
```

Comment:
why is this needed

Reply:
It's a new company ability. The ability to (at any point during the round, but that's handled via the normal action timing) purchase the current train. I added a readme item here: https://github.com/kelsin/18xx/blob/g_1871/lib/engine/ability/README.md#purchase_train